core/src/conv.c

```diff
@@ -1,3 +1,5 @@
 #include "conv.h"
 
-uint64_t conv_btc_to_satoshi(uint32_t btc) { return btc * 100000000; }
+uint64_t conv_btc_to_satoshi(uint32_t btc) {
+    return (uint64_t) btc * 100000000;
```

Comment:
This is the actual bug fix - if we don't cast btc to a 64-bit uint before the multiplication, we end up doing 32-bit * 32-bit multiplication -> 32-bit result. This overflows and truncates the result, then casts the truncated result into the 64-bit return type. So, the cast to 64-bit has to happen before the multiplication.